hdf/src/mfgr.c

```diff
@@ -1282,7 +1277,7 @@ GRIget_image_list(int32 file_id, gr_info_t *gr_ptr)
         }     /* end if */
     }         /* end for */
 
-    HDfree(img_info); /* free image info structures */
+    free(img_info); /* free image info structures */
```

Comment:
If this was moved to below "done" then the frees for every error would not be needed.